### DISCUSSION TITLE
Pantry Ingredients list

### DISCUSSION THREAD

**Discussion by yeastie** [mealie-recipes/mealie#2448] [Feature Request]
I have done a search and can't find anything explicitly requesting this feature (Or I am really bad at searching!). I think it would be a great idea to be able to keep track of our pantry/freezer ingredients in the app, to make it easier to identify what might need to go on the shopping list when looking at recipes

**Comment by danieledwardgeorgehitchcock** [comment]
Got my vote! (was going to raise a similar request) - it would also be great for mealie to suggest recipes based on current inventory (for those just before paycheque days)

**Comment by Madj42** [comment]
Was looking for the same thing.  I would be happy with something similar to the shopping list function as a stop gap.  We already have the ingredients capability.  Could just build off of this and allow us to say how much of each we have in stock.  This doesn't have to be the level that grocy is at for me.  Just something to track would be extremely helpful.

**Comment by enrauk** [comment]
Yes, this would be a great feature especially if items are marked as bought on the shopping list are transfered to inventory

**Comment by michael-genson** [comment]
We recently added an "on hand" flag to foods which automatically deselects the food when adding a recipe to your shopping list:
![image](https://github.com/user-attachments/assets/7c05268f-5604-46c0-9421-0c0aad141227)

While not as robust as what you're asking for, it achieves the same thing with shopping lists

**Reply by almulder** [reply]
I would like to expand on this and have it that we can just enter in a list with as many or as few ingredients as we want and it tells us what we can make based on that. Like I have x,y,z and i dont know what to make, let the app show me what can be made with the info provided and be able to sort said list based on how many of the ingredients in the list can be used, '

so say i enter in chicken breasts and ground beef along with other ingredients, the results would list all the recipes that can be made with said ingredients  listing from the most ingredients to least , and even a refine list so i could check chicken and ingredient x,z and it would show me the ones that use those, or if i select ground beef and zy,z and it would show me ones that use them. 

This would really make it flexible for meal prepping. And as someone who buys a full cow butchered and full pig butchered. would be great to put in a cut of meat and see what my options are.

**Comment by cloudbr34k84** [comment]
yep this would be great!

**Comment by ekarious** [comment]
This is actually one of the feature I'm really looking on. Managing my kitchen ingredients

**Comment by cloudbr34k84** [comment]
It would be great to import grocery orders as ingredients etc

**Comment by XJonnyP** [comment]
I too would love the ability to track what I have on hand.

**Reply by Kuchenpirat** [reply]
This is something we probably won't get into it, because the constant manual upkeep required to keep this up to date is something we don't want mealie to represent. 
If you wan't a programm that can handle those requirements you might want to take a look at [grocy](https://github.com/grocy/grocy).

**Reply by swcarter007** [reply]
That makes sense from a direction standpoint.  What about a potential integration hook that you can use to call something like grocy to provide the ingredients list back to mealie for the recipe.  So you don't necessarily have to build in the maintenance, but can still potentially benefit from being able to get a list of ingredients?

**Comment by ScuttleShip** [comment]
Would love a proper pantry tracker built into this where it removes ingredients on recipes I make and I can go through my stock and say what I have in.

**Comment by junelac** [comment]
I totally think it would be a great addition, because it would allows a killer feature: proposing/filtering  recipes based on what is available at home.  AFAIK there is no self-hosted recipe manager that propose this. 

I know that grocy exists but it's much too convoluted for an individual/family usage.  

As for the upkeep, there are solutions to ease it. For adding ingredients and items, providing a database of common ingredients (that could even be updated with recipes and shared betwwen users) and a tool like barcode buddy  from grocy would do a lot. For removing ingredients, there is no need to be very precise: knowing if I still have half of the frozen spinash pack or if I have any amount of salt is more than enough, there is no need to check for the exact quantity remaining.

(note that I'm  not saying you should do it, I'm just sharing my thought on the topic because I recently searched a solution)

**Reply by michael-genson** [reply]
This is actually already supported: https://github.com/mealie-recipes/mealie/pull/4542
The "on hand" flag can be used to globally mark something as available or not

**Comment by idunwannagotoschool** [comment]
I agree! Something like grocy would be fantastic so theres a dedicated page to track stock instead of having to click into each food which is not the most convenient thing to do.
